tests/THTests.hs

```diff
@@ -0,0 +1,31 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell   #-}
+
+module THTests (tests) where
+
+import qualified Data.Set         as S
+import           Data.Text        (Text)
+import           Nix
+import           Nix.TH
+import           Test.Tasty
+import           Test.Tasty.HUnit
+import           Test.Tasty.TH
+
+-- Regression test for #373
+case_regression_373 :: Assertion
+case_regression_373 = do
+  freeVarsEqualTest "{ inherit a; }" ["a"]
+  freeVarsEqualTest "rec {inherit a; }" ["a"]
+  freeVarsEqualTest "let inherit a; in { }" ["a"]
```

Comment:
There is some fancy TemplateHaskell thing to have Nix expressions directly without going through strings, but I don't know how it works